### PR TITLE
Spike: Task to load data into BigQuery.

### PR DESCRIPTION
**Pull request by HassanJaveed84**
Added Luigi task to load data from S3 into BigQuery.

- Supports Loading from the result of an input task to BigQuery
- Supports overwrite

Issues:
If you do a streaming insert into a BigQuery table(I do that for 'table_updates'), you lose the ability to run DML statements on that table(specifically DELETE statements in our case) until the streaming buffer is unset( 

@mulby @brianhw @azafty468 @tobz

### CODE REVIEW

**Comment by mulby** [comment]
@HassanJaveed84 I was hoping we could use their "load from file" HTTP based loading method like this: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/bigquery/cloud-client/load_data_from_file.py

More info here: https://cloud.google.com/bigquery/loading-data-post-request#bigquery-import-file-python

I'm worried about us having to clean up the files on GCS after we are done loading. Not terribly difficult I suppose, but would like to avoid the complexity if possible...

**Comment by HassanJaveed84** [comment]
Initially I was thinking of using the post method but avoided trying it out as there are a bunch of requirements for the file object passed. Going to try to see if it works.

**Comment by mulby** [comment]
Yeah it would be awesome if we could pass it a luigi target:

```
with self.input().open('r') as input_file:
    job = table.upload_from_file(
            input_file, source_format='text/csv')
```

**Comment by HassanJaveed84** [comment]
I've updated and tested it with an ExternalUrl type as the insert source task. Other input tasks should behave the same way.  
Great suggestion!

**Review by azafty468 — APPROVED** [review]
👍 I'm not familiar with the ins-and-outs of Big Query but the general structure looks good to me

File: edx/analytics/tasks/common/bigquery_load.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        client = self.output().client
+        self.create_dataset(client)
+        self.init_copy(client)
+        self.create_table(client)
+
+        dataset = client.dataset(self.dataset_id)
+        table = dataset.table(self.table, self.schema)
+
+        with self.input()['source'].open('r') as source_file:
+            job = table.upload_from_file(source_file, source_format='text/csv', field_delimiter=self.field_delimiter)
```
Comment (brianhw) [inline edx/analytics/tasks/common/bigquery_load.py:136]: Just out of curiosity, does this load our TSV data from S3?  I'm assuming we would need a way to specify the field_delimiter, to allow changing the value from a comma to a tab.

**Pull request closed without merging**